discovery/gossiper_test.go

```diff
@@ -847,9 +1236,319 @@ func TestOrphanSignatureAnnouncement(t *testing.T) {
+	}
+
+	// Shut down gossiper, and restart. This should trigger a new attempt
+	// to send the message to the peer.
+	ctx.gossiper.Stop()
+	gossiper, err := New(Config{
+		Notifier:  ctx.gossiper.cfg.Notifier,
+		Broadcast: ctx.gossiper.cfg.Broadcast,
+		SendToPeer: func(target *btcec.PublicKey,
+			msg ...lnwire.Message) error {
+			return fmt.Errorf("intentional error in SendToPeer")
+		},
+		NotifyWhenOnline: func(peer *btcec.PublicKey,
+			connectedChan chan<- struct{}) {
+			notifyPeers <- connectedChan
+		},
+		Router:           ctx.gossiper.cfg.Router,
+		TrickleDelay:     trickleDelay,
+		RetransmitDelay:  retransmitDelay,
+		ProofMatureDelta: proofMatureDelta,
+		DB:               ctx.gossiper.cfg.DB,
+	}, ctx.gossiper.selfKey)
+	if err != nil {
+		t.Fatalf("unable to recreate gossiper: %v", err)
+	}
+	if err := gossiper.Start(); err != nil {
+		t.Fatalf("unable to start recreated gossiper: %v", err)
+	}
+	defer gossiper.Stop()
+
+	ctx.gossiper = gossiper
+
+	// After starting up, the gossiper will see that it has a waitingproof
+	// in the database, and will retry sending its part to the remote. Since
+	// SendToPeer will fail again, it should register for a notification
+	// when the peer comes online.
+	select {
+	case conChan = <-notifyPeers:
+	case <-time.After(2 * time.Second):
+		t.Fatalf("gossiper did not ask to get notified when " +
+			"peer is online")
+	}
+
+	// Fix the SendToPeer method.
+	sentToPeer := make(chan lnwire.Message, 1)
+	ctx.gossiper.cfg.SendToPeer = func(target *btcec.PublicKey,
+		msg ...lnwire.Message) error {
+		sentToPeer <- msg[0]
+		return nil
+	}
+	// Notify that peer is now online. This should trigger a new call
+	// to SendToPeer.
+	close(conChan)
+
+	select {
+	case <-sentToPeer:
+	case <-time.After(2 * time.Second):
+		t.Fatalf("gossiper did not send message when peer came online")
+	}
+
+	// Now exchanging the remote channel proof, the channel annoncement
+	// broadcast should continue as normal.
+	select {
+	case err = <-ctx.gossiper.ProcessRemoteAnnouncement(batch.remoteProofAnn,
+		remoteKey):
+	case <-time.After(2 * time.Second):
+		t.Fatal("did not process remote announcement")
+	}
+	if err != nil {
+		t.Fatalf("unable to process :%v", err)
+	}
+
+	for i := 0; i < 3; i++ {
+		select {
+		case <-ctx.broadcastedMessage:
+		case <-time.After(time.Second):
+			t.Fatal("announcement wasn't broadcast")
+		}
+	}
+
+	number = 0
+	if err := ctx.gossiper.waitingProofs.ForAll(
+		func(*channeldb.WaitingProof) error {
+			number++
+			return nil
+		},
+	); err != nil && err != channeldb.ErrWaitingProofNotFound {
+		t.Fatalf("unable to retrieve objects from store: %v", err)
+	}
+
 	if number != 0 {
+		t.Fatal("waiting proof should be removed from storage")
+	}
+}
+
+// Test that we retry sending the AnnounceSignatures to the remote peer,
+// even though we have been able to receive both parts of the proof.
+func TestSignatureAnnouncementRetryAfterBroadcast(t *testing.T) {
```

Comment:
Doesn't seem like this is needed. If we have the full proof (a fully signed chan ann message), then we don't need to send it to the remote peer, as upon reconnect, we'll just send them the full proof (which is what we're trying to create anyway).